File: mobile/src/main/java/com/windscribe/mobile/account/AccountView.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.mobile.account;

public interface AccountView {


    void goToConfirmEmailActivity();

    void goToEmailActivity();

    void hideProgress();

    void openEditAccountInBrowser(String url);

    void openUpgradeActivity();

    void setActivityTitle(String title);

    void setEmail(String email, int textColor, int labelColor);

    void setEmailConfirm(String emailConfirm, String warningText, int emailColor, int emailLabelColor, int infoIcon, int containerBackground);

    void setEmailConfirmed(String emailConfirm, String warningText, int emailColor, int emailLabelColor, int infoIcon, int containerBackground);

    void setPlanName(String planName);

    void setDataLeft(String dataLeft);

    void setResetDate(String resetDateLabel, String resetDate);

    void setUsername(String username);

    void setWebSessionLoading(boolean show);

    void setupLayoutForFreeUser(String upgradeText);

    void setupLayoutForGhostMode(boolean proUser);

    void setupLayoutForPremiumUser(String upgradeText);

    void showEnterCodeDialog();

    void showErrorDialog(String error);

    void showErrorMessage(String errorMessage);

    void showProgress(String progressText);

    void showSuccessDialog(String message);
}

File: mobile/src/main/java/com/windscribe/mobile/account/AccountView.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */
package com.windscribe.mobile.account

interface AccountView {
    fun goToConfirmEmailActivity()
    fun goToEmailActivity()
    fun hideProgress()
    fun openEditAccountInBrowser(url: String)
    fun openUpgradeActivity()
    fun setActivityTitle(title: String)
    fun setEmail(
        email: String,
        warningText: String,
        emailColor: Int,
        warningColor: Int,
        labelColor: Int,
        infoIcon: Int,
        containerBackground: Int
    )

    fun setEmailConfirm(
        emailConfirm: String,
        warningText: String,
        emailColor: Int,
        emailLabelColor: Int,
        infoIcon: Int,
        containerBackground: Int
    )

    fun setEmailConfirmed(
        emailConfirm: String,
        warningText: String,
        emailColor: Int,
        emailLabelColor: Int,
        infoIcon: Int,
        containerBackground: Int
    )

    fun setPlanName(planName: String)
    fun setDataLeft(dataLeft: String)
    fun setResetDate(resetDateLabel: String, resetDate: String)
    fun setUsername(username: String)
    fun setWebSessionLoading(show: Boolean)
    fun setupLayoutForGhostMode(proUser: Boolean)
    fun setupLayoutForFreeUser(upgradeText: String, color: Int)
    fun setupLayoutForPremiumUser(upgradeText: String, color: Int)
    fun showEnterCodeDialog()
    fun showErrorDialog(error: String)
    fun showErrorMessage(errorMessage: String)
    fun showProgress(progressText: String)
    fun showSuccessDialog(message: String)
    fun showEnterVoucherCodeDialog()
}